torch/utils/collect_env.py

```diff
@@ -243,6 +269,152 @@ def get_nvidia_smi():
     return smi
 
 
+def _detect_linux_pkg_manager():
+    if get_platform() != "linux":
+        return "N/A"
+    for mgr_name in ["dpkg", "dnf", "yum", "zypper"]:
+        rc, _, _ = run(f"which {mgr_name}")
+        if rc == 0:
+            return mgr_name
+    return "N/A"
+
+
+def get_linux_pkg_version(run_lambda, pkg):
+    pkg_mgr = _detect_linux_pkg_manager()
+    if pkg_mgr == "N/A":
+        return "N/A"
+
+    pkgs = {
```

Comment:
I don't think this list belongs to this function, as this should be usable to get version of any linux package, not just those two